### PR TITLE
added null check to prevent NPE

### PR DESCRIPTION
**Pull request by cafesilencio**
## Description
#2807 identifies a NPE when update location in NavigationMapboxMap is called with a null value.

- [x] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [x] I have added the appropriate milestone and project boards

### Implementation
A null check is done on the incoming location parameter. 

## Testing

- [x] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [x] I have tested via a test drive, or a simulation/mock location app
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/map/NavigationMapboxMap.java [hunk 1]
```diff
@@ -236,9 +236,11 @@ public void clearMarkers() {
    * @param location to update the icon and query the map
    */
   public void updateLocation(Location location) {
```
Comment (Guardiola31337) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/map/NavigationMapboxMap.java:238]: As this is a `public` method, could we add `@NonNull` / `@Nullable` annotations for Kotlin users 😅?

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/map/NavigationMapboxMap.java [hunk 2]
```diff
@@ -235,10 +235,12 @@ public void clearMarkers() {
    *
    * @param location to update the icon and query the map
    */
-  public void updateLocation(Location location) {
-    List<Location> locations = new ArrayList<>(1);
-    locations.add(location);
-    updateLocation(locations);
+  public void updateLocation(@Nullable Location location) {
```
Comment (Guardiola31337) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/map/NavigationMapboxMap.java:238]: Should it be `@NonNull` 🤔?

```suggestion
  public void updateLocation(@NonNull Location location) {
```

Comment (cafesilencio) [reply libnavigation-ui/src/main/java/com/mapbox/navigation/ui/map/NavigationMapboxMap.java]: d'oh! that's what I get for being in a hurry.

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/map/NavigationMapboxMap.java [hunk 3]
```diff
@@ -235,10 +235,12 @@ public void clearMarkers() {
    *
    * @param location to update the icon and query the map
    */
-  public void updateLocation(Location location) {
-    List<Location> locations = new ArrayList<>(1);
-    locations.add(location);
-    updateLocation(locations);
+  public void updateLocation(@NonNull Location location) {
+    if (location != null) {
```
Comment (RingerJK) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/map/NavigationMapboxMap.java:239]: sorry, `@NonNull` but we check it on null. should be `@Nullable`? 😄

Comment (RingerJK) [reply libnavigation-ui/src/main/java/com/mapbox/navigation/ui/map/NavigationMapboxMap.java:239]: or don't check if is `null`

Comment (korshaknn) [reply libnavigation-ui/src/main/java/com/mapbox/navigation/ui/map/NavigationMapboxMap.java:239]: yep, it's confusing to check `@NonNull` nullability
btw, I get compilation errors with calling that method. Have you checked that it compiles?

**Comment by korshaknn** [comment]
Screen Shot 2020-04-21 at 12 28 06

**Comment by RingerJK** [comment]
need fix, ui test failed on examples build https://app.circleci.com/pipelines/github/mapbox/mapbox-navigation-android/5776/workflows/b97d8712-d5c5-49ce-af50-05385c18bd77

**Comment by RingerJK** [comment]
@cafesilencio https://github.com/mapbox/mapbox-navigation-android/pull/2815/files#diff-e5a5b815f804b7277b6f60237aacc739R238

**Comment by korshaknn** [comment]
fixed in [PR 2815](https://github.com/mapbox/mapbox-navigation-android/pull/2815)

**Pull request merged**